code/modules/projectiles/guns/projectile/sniper.dm

```diff
@@ -79,11 +80,60 @@
 	set name = "Use Scope"
 	set popup_menu = 1
 
+	if(!has_scope)
+		to_chat(usr, SPAN_WARNING("\The [src] doesn't have a scope!"))
+		return
+
 	if(wielded)
 		toggle_scope(2.0, usr)
 	else
 		to_chat(usr, "<span class='warning'>You can't look through the scope without stabilizing the rifle!</span>")
 
+/obj/item/gun/projectile/heavysniper/unathi
+	name = "hegemony slugger"
+	desc = "An incredibly large firearm, produced by an Ouerean Guild. Uses custom slugger rounds."
+	icon = 'icons/obj/guns/unathi_slugger.dmi'
+	icon_state = "slugger"
+	item_state = "slugger"
+	origin_tech = list(TECH_COMBAT = 8, TECH_MATERIAL = 2, TECH_ILLEGAL = 4)
+	w_class = ITEMSIZE_HUGE
+	fire_sound = 'sound/effects/Explosion1.ogg'
+	caliber = "slugger"
+	ammo_type = /obj/item/ammo_casing/slugger
+	magazine_type = null
+	has_scope = FALSE
+
+/obj/item/gun/projectile/heavysniper/unathi/update_icon()
+	if(bolt_open && length(loaded))
+		icon_state = "slugger-open-loaded"
+	else if(bolt_open && !length(loaded))
+		icon_state = "slugger-open"
+	else
+		icon_state = "slugger"
+	if(wielded)
+		item_state = "slugger-wielded"
+	else
+		item_state = "slugger"
+	update_held_icon()
+
+/obj/item/gun/projectile/heavysniper/unathi/handle_post_fire(mob/user)
+	..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.mob_size < 10) // smaller than an unathi
```

Comment:
Imho it would be good to check for rigs here, so you can ignore the effects with a active and deployed rig